express-delete/index.js

```diff
@@ -0,0 +1,42 @@
+const express = require('express');
+let grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+}
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  let reply = [];
+  for (prop in grades) {
+    reply.push(grades[prop]);
+  }
+  res.json(reply);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  let gradeID = [];
+  for (prop in grades) gradeID.push(prop);
+  delete grades[gradeID[req.params.id]];
+  res.sendStatus(204);
+})
```

Comment:
The `:id` in the URL should delete a grade by matching `id` property. In this example, `DELETE /api/grades/2` should not delete anything (there is no grade with an `id` of `2`), while `DELETE /api/grades/273` should remove Forbes Lindsay's grade:

---

![image](https://user-images.githubusercontent.com/16143771/128807651-07460a5f-8c06-45e8-9e81-88d1dd32f2da.png)

Reply:
I thought so.... overthought it